spark/src/main/scala/org/apache/spark/sql/delta/commands/DescribeDeltaHistoryCommand.scala

```diff
@@ -17,85 +17,104 @@
+    DescribeDeltaHistory(plan, limit)
+  }
+
+  val COMMAND_NAME = "DESCRIBE HISTORY"
+}
+
 /**
  * A logical placeholder for describing a Delta table's history, so that the history can be
  * leveraged in subqueries. Replaced with `DescribeDeltaHistoryCommand` during planning.
  *
  * @param options: Hadoop file system options used for read and write.
  */
 case class DescribeDeltaHistory(
-    path: Option[String],
-    tableIdentifier: Option[TableIdentifier],
+    override val child: LogicalPlan,
     limit: Option[Int],
-    options: Map[String, String],
-    output: Seq[Attribute] = ExpressionEncoder[DeltaHistory]().schema.toAttributes)
-  extends LeafNode with MultiInstanceRelation  {
-  override def computeStats(): Statistics = Statistics(sizeInBytes = conf.defaultSizeInBytes)
+    override val output: Seq[Attribute] = ExpressionEncoder[DeltaHistory]().schema.toAttributes)
+  extends UnaryNode
+    with MultiInstanceRelation
+    with DeltaCommand {
 
   override def newInstance(): LogicalPlan = copy(output = output.map(_.newInstance()))
+
+  override def withNewChildInternal(newChild: LogicalPlan): LogicalPlan = copy(child = newChild)
+
+  /**
+   * Define this operator as having no attributes provided by children in order to prevent column
+   * pruning from trying to insert projections above the source relation.
+   */
+  override lazy val references: AttributeSet = AttributeSet.empty
+  override def inputSet: AttributeSet = AttributeSet.empty
+  assert(!child.isInstanceOf[Project],
+    s"The child operator of DescribeDeltaHistory must not contain any projection: $child")
+
+  /** Converts this operator into an executable command. */
+  def toCommand: DescribeDeltaHistoryCommand = {
+    // Max array size
+    if (limit.exists(_ > Int.MaxValue - 8)) {
+      throw DeltaErrors.maxArraySizeExceeded()
+    }
+    val deltaTableV2: DeltaTableV2 = getDeltaTable(child, DescribeDeltaHistory.COMMAND_NAME)
+    DescribeDeltaHistoryCommand(
+      table = deltaTableV2,
+      limit = limit,
+      output = output)
+  }
 }
 
 /**
  * A command for describing the history of a Delta table.
- *
- * @param options: Hadoop file system options used for read and write.
  */
 case class DescribeDeltaHistoryCommand(
-    path: Option[String],
-    tableIdentifier: Option[TableIdentifier],
+    @transient table: DeltaTableV2,
     limit: Option[Int],
-    options: Map[String, String],
     override val output: Seq[Attribute] = ExpressionEncoder[DeltaHistory]().schema.toAttributes)
-  extends LeafRunnableCommand with DeltaLogging {
-
-  override def run(sparkSession: SparkSession): Seq[Row] = {
-    val basePath =
-      if (path.nonEmpty) {
-        new Path(path.get)
-      } else if (tableIdentifier.nonEmpty) {
-        val sessionCatalog = sparkSession.sessionState.catalog
-        lazy val metadata = sessionCatalog.getTableMetadata(tableIdentifier.get)
+  extends LeafRunnableCommand
+    with MultiInstanceRelation
+    with DeltaCommand
```

Comment:
Why does this need to become a `DeltaCommand`? What functionality does it use?

Reply:
Good point, we should only need `DeltaLogging` here for `recordDeltaOperation`. Will change.